### PR TITLE
move cloud-providers to top go module

### PR DESCRIPTION
**Pull request by liudalibj**
Remove cyclic dependency among cloud-api-adaptor and peerpod-ctrl.
Move cloud-providers to a new go mod and re-structure all the source codes as:
```
cloud-api-adaptor
    \src\cloud-api-adaptor
         cloud-providers
         peerpod-ctl
         peerpodcfg-ctl
         webhook
         csi-wrapper
```
And updated all workflow yamls to reflect new directory structure.

part of https://github.com/confidential-containers/cloud-api-adaptor/issues/1122

Signed-off-by: Da Li Liu <liudali@cn.ibm.com>
Co-authored-by: James Tumber <james.tumber@ibm.com>

### CODE REVIEW

**Comment by liudalibj** [comment]
The pr are ready to review.
The `est_e2e_libvirt` test didn't passed, it failed to build podvm images.
From https://github.com/confidential-containers/cloud-api-adaptor/actions/runs/8358226935/job/22879052517?pr=1719#step:1:38 
```
Uses: confidential-containers/cloud-api-adaptor/.github/workflows/podvm_builder.yaml@refs/heads/main (b499f84492c1ad4e2755d04ef31b10532e9a272c)
```
it seems that it use the workflow file from main branch, it should use the podvm build yaml from this PR.
I am trying to find out how to let the GHA using podvm build yaml from this pr.

**Comment by liudalibj** [comment]
- tried use ref from the pr, but failed:
from [Invalid workflow file: .github/workflows/e2e_run_all.yaml#L44](https://github.com/liudalibj/cloud-api-adaptor/actions/runs/8358842819/workflow)
invalid value workflow reference: cannot specify version when calling local workflows

- tried make the yaml files in main branch first and then use pr to test, the issue is gone
https://github.com/liudalibj/cloud-api-adaptor/actions/runs/8358963893/job/22881414028

So we maybe need merge this pr first without test the `test_e2e_libvirt` result, when all related yamls in main then can test it from a new pr.

**Comment by stevenhorsman** [comment]
@liudalibj - so the problem is that the workflow files comes from the `main` branch, not this PR. However we might be able to get around this as I think that things like `hack/ci-helper.sh` (and probably quite a bit of the `hack` directory) are not related to the cloud-api-adaptor go module, so I think they can stay in the root of the repo, rather than moving into `/src/cloud-api-adaptor`. I think this is also the case for things like docs and the podvm directories as well? Maybe this is something we can discuss on the call later?

**Comment by liudalibj** [comment]
> @liudalibj - so the problem is that the workflow files comes from the `main` branch, not this PR. However we might be able to get around this as I think that things like `hack/ci-helper.sh` (and probably quite a bit of the `hack` directory) are not related to the cloud-api-adaptor go module, so I think they can stay in the root of the repo, rather than moving into `/src/cloud-api-adaptor`. I think this is also the case for things like docs and the podvm directories as well? Maybe this is something we can discuss on the call later?

@stevenhorsman base on this comment, I move some common scripts(check lint...) to the root hack folder, and also create new `hack/release-helper.sh` to help do git tags in the release process.

**Comment by liudalibj** [comment]
And verified that the `test_e2e_libvirt` will only work after we have the PR in main branch.
Here is the verify PR https://github.com/liudalibj/cloud-api-adaptor/pull/3

**Comment by liudalibj** [comment]
- Create one new Z VSI and follow https://github.com/liudalibj/cloud-api-adaptor/tree/toplevel-cloud-module/src/cloud-api-adaptor/libvirt document.

- The built out test CAA image and podvm image from my repo work as expected:
  - quay.io/liudalibj/cloud-api-adaptor:dev-7d3fdfc81246c603c9bc70a3c5d34a82f832ae2a
  - quay.io/liudalibj/podvm-generic-ubuntu-s390x:95c13d7283bfd3b9f8a4db8d45850973144d0a6b
```
root@liudali-s390x-libvirt:~/cloud-api-adaptor/src/cloud-api-adaptor# kubectl get po -A
NAMESPACE                        NAME                                               READY   STATUS             RESTARTS      AGE
confidential-containers-system   cc-operator-controller-manager-857f844f7d-qbdx8    2/2     Running            1 (26m ago)   60m
confidential-containers-system   cc-operator-daemon-install-95hrv                   1/1     Running            0             59m
confidential-containers-system   cc-operator-pre-install-daemon-sfh67               1/1     Running            0             60m
confidential-containers-system   cloud-api-adaptor-daemonset-z4vsz                  1/1     Running            0             60m
confidential-containers-system   peerpod-ctrl-controller-manager-74c65cd59f-6hbsb   2/2     Running            1 (26m ago)   60m
default                          nginx                                              1/1     Running            0             25m
ingress-nginx                    ingress-nginx-admission-create-hpxzc               0/1     Completed          0             91m
ingress-nginx                    ingress-nginx-admission-patch-gltxq                0/1     Completed          2             91m
ingress-nginx                    ingress-nginx-controller-7bf7bc78dc-qcs8r          0/1     ImagePullBackOff   0             91m
kube-flannel                     kube-flannel-ds-2ldqk                              1/1     Running            0             89m
kube-flannel                     kube-flannel-ds-5c76z                              1/1     Running            0             91m
kube-system                      coredns-787d4945fb-7pzlt                           1/1     Running            0             91m
kube-system                      coredns-787d4945fb-mkp7l                           1/1     Running            0             91m
kube-system                      etcd-peer-pods-ctlplane-0                          1/1     Running            0             91m
kube-system                      kube-apiserver-peer-pods-ctlplane-0                1/1     Running            0             91m
kube-system                      kube-controller-manager-peer-pods-ctlplane-0       1/1     Running            2 (26m ago)   91m
kube-system                      kube-proxy-jwc5d                                   1/1     Running            0             89m
kube-system                      kube-proxy-x8jqw                                   1/1     Running            0             91m
kube-system                      kube-scheduler-peer-pods-ctlplane-0                1/1     Running            2 (26m ago)   91m
root@liudali-s390x-libvirt:~/cloud-api-adaptor/src/cloud-api-adaptor#
```
```
kubectl logs -f -n confidential-containers-system   cloud-api-adaptor-daemonset-z4vsz
+ exec cloud-api-adaptor libvirt -uri 'qemu+ssh://root@192.168.122.1/system?no_verify=1' -data-dir /opt/data-dir -pods-dir /run/peerpod/pods -network-name default -pool-name default -disable-cvm -socket /run/peerpod/hypervisor.sock
cloud-api-adaptor version v0.9.0.alpha.1-dev
  commit: 7d3fdfc81246c603c9bc70a3c5d34a82f832ae2a
  go: go1.21.8
cloud-api-adaptor: starting Cloud API Adaptor daemon for "libvirt"
2024/03/21 07:22:02 [adaptor/cloud/libvirt] libvirt config: &libvirt.Config{URI:"qemu+ssh://root@192.168.122.1/system?no_verify=1", PoolName:"default", NetworkName:"default", DataDir:"/opt/data-dir", DisableCVM:true, VolName:"podvm-base.qcow2", LaunchSecurity:"", Firmware:"/usr/share/edk2/ovmf/OVMF_CODE.fd"}
2024/03/21 07:22:03 [adaptor/cloud/libvirt] Created libvirt connection
2024/03/21 07:22:03 [adaptor] server config: &adaptor.ServerConfig{TLSConfig:(*tlsutil.TLSConfig)(0xc000511180), SocketPath:"/run/peerpod/hypervisor.sock", CriSocketPath:"", PauseImage:"", PodsDir:"/run/peerpod/pods", ForwarderPort:"15150", ProxyTimeout:300000000000, AAKBCParams:"", EnableCloudConfigVerify:false}
2024/03/21 07:22:03 [util/k8sops] initialized PeerPodService
2024/03/21 07:22:03 [probe/probe] Using port: 8000
2024/03/21 07:22:03 [adaptor] server started
2024/03/21 07:22:25 [probe/probe] nodeName: peer-pods-worker-0
2024/03/21 07:22:25 [probe/probe] Selected pods count: 9
2024/03/21 07:22:25 [probe/probe] Ignored standard pod: cc-operator-controller-manager-857f844f7d-qbdx8
2024/03/21 07:22:25 [probe/probe] Ignored standard pod: cc-operator-pre-install-daemon-sfh67
2024/03/21 07:22:25 [probe/probe] Ignored standard pod: cloud-api-adaptor-daemonset-z4vsz
2024/03/21 07:22:25 [probe/probe] Ignored standard pod: peerpod-ctrl-controller-manager-74c65cd59f-6hbsb
2024/03/21 07:22:25 [probe/probe] Ignored standard pod: ingress-nginx-admission-create-hpxzc
2024/03/21 07:22:25 [probe/probe] Ignored standard pod: ingress-nginx-admission-patch-gltxq
2024/03/21 07:22:25 [probe/probe] Ignored standard pod: ingress-nginx-controller-7bf7bc78dc-qcs8r
2024/03/21 07:22:25 [probe/probe] Ignored standard pod: kube-flannel-ds-2ldqk
2024/03/21 07:22:25 [probe/probe] Ignored standard pod: kube-proxy-jwc5d
2024/03/21 07:22:25 [probe/probe] All PeerPods standup. we do not check the PeerPods status any more.
2024/03/21 07:57:18 [podnetwork] routes on netns /var/run/netns/cni-aca249c0-cf2a-69d2-ae44-be82d82f22ea
2024/03/21 07:57:18 [podnetwork]     0.0.0.0/0 via 10.244.1.1 dev eth0
2024/03/21 07:57:18 [podnetwork]     10.244.0.0/16 via 10.244.1.1 dev eth0
2024/03/21 07:57:18 [adaptor/cloud] Credentials file is not in a valid Json format, ignored
2024/03/21 07:57:18 [adaptor/cloud] stored /run/peerpod/pods/dc5d14f97adad43f08e9d30c8e93d400193a4b4904786f4776449e693a4dc847/daemon.json
2024/03/21 07:57:18 [adaptor/cloud] create a sandbox dc5d14f97adad43f08e9d30c8e93d400193a4b4904786f4776449e693a4dc847 for pod nginx in namespace default (netns: /var/run/netns/cni-aca249c0-cf2a-69d2-ae44-be82d82f22ea)
2024/03/21 07:57:18 [adaptor/cloud/libvirt] LaunchSecurityType: None
2024/03/21 07:57:18 [adaptor/cloud/libvirt] Checking if instance (podvm-nginx-dc5d14f9) exists
2024/03/21 07:57:18 [adaptor/cloud/libvirt] Uploaded volume key /var/lib/libvirt/images/podvm-nginx-dc5d14f9-root.qcow2
2024/03/21 07:57:18 [adaptor/cloud/libvirt] Create cloudInit iso
2024/03/21 07:57:18 [adaptor/cloud/libvirt] Uploading iso file: podvm-nginx-dc5d14f9-cloudinit.iso
2024/03/21 07:57:18 [adaptor/cloud/libvirt] 45056 bytes uploaded
2024/03/21 07:57:18 [adaptor/cloud/libvirt] Volume ID: /var/lib/libvirt/images/podvm-nginx-dc5d14f9-cloudinit.iso
2024/03/21 07:57:18 [adaptor/cloud/libvirt] Create XML for 'podvm-nginx-dc5d14f9'
2024/03/21 07:57:18 [adaptor/cloud/libvirt] Creating VM 'podvm-nginx-dc5d14f9'
2024/03/21 07:57:18 [adaptor/cloud/libvirt] Starting VM 'podvm-nginx-dc5d14f9'
2024/03/21 07:57:20 [adaptor/cloud/libvirt] VM id 290
2024/03/21 07:57:41 [adaptor/cloud/libvirt] Instance created successfully
2024/03/21 07:57:41 [adaptor/cloud/libvirt] created an instance podvm-nginx-dc5d14f9 for sandbox dc5d14f97adad43f08e9d30c8e93d400193a4b4904786f4776449e693a4dc847
2024/03/21 07:57:41 [util/k8sops] nginx is now owning a PeerPod object
2024/03/21 07:57:41 [adaptor/cloud] created an instance podvm-nginx-dc5d14f9 for sandbox dc5d14f97adad43f08e9d30c8e93d400193a4b4904786f4776449e693a4dc847
2024/03/21 07:57:41 [tunneler/vxlan] vxlan ppvxlan1 (remote 192.168.122.103:4789, id: 555000) created at /proc/1/task/12/ns/net
2024/03/21 07:57:41 [tunneler/vxlan] vxlan ppvxlan1 created at /proc/1/task/12/ns/net
2024/03/21 07:57:41 [tunneler/vxlan] vxlan ppvxlan1 is moved to /var/run/netns/cni-aca249c0-cf2a-69d2-ae44-be82d82f22ea
2024/03/21 07:57:41 [tunneler/vxlan] Add tc redirect filters between eth0 and vxlan1 on pod network namespace /var/run/netns/cni-aca249c0-cf2a-69d2-ae44-be82d82f22ea
2024/03/21 07:57:41 [adaptor/proxy] Listening on /run/peerpod/pods/dc5d14f97adad43f08e9d30c8e93d400193a4b4904786f4776449e693a4dc847/agent.ttrpc
2024/03/21 07:57:41 [adaptor/proxy] failed to init cri client, the err: cri runtime endpoint is not specified, it is used to get the image name from image digest
2024/03/21 07:57:41 [adaptor/proxy] Trying to establish agent proxy connection to 192.168.122.103:15150
2024/03/21 07:57:41 [adaptor/proxy] established agent proxy connection to 192.168.122.103:15150
2024/03/21 07:57:41 [adaptor/cloud] agent proxy is ready
2024/03/21 07:57:41 [adaptor/proxy] CreateSandbox: hostname:nginx sandboxId:dc5d14f97adad43f08e9d30c8e93d400193a4b4904786f4776449e693a4dc847
2024/03/21 07:57:41 [adaptor/proxy]     storages:
2024/03/21 07:57:41 [adaptor/proxy]         mountpoint:/run/kata-containers/sandbox/shm source:shm fstype:tmpfs driver:ephemeral
2024/03/21 07:57:41 [adaptor/proxy] CreateContainer: containerID:dc5d14f97adad43f08e9d30c8e93d400193a4b4904786f4776449e693a4dc847
2024/03/21 07:57:41 [adaptor/proxy]     mounts:
2024/03/21 07:57:41 [adaptor/proxy]         destination:/proc source:proc type:proc
2024/03/21 07:57:41 [adaptor/proxy]         destination:/dev source:tmpfs type:tmpfs
2024/03/21 07:57:41 [adaptor/proxy]         destination:/dev/pts source:devpts type:devpts
2024/03/21 07:57:41 [adaptor/proxy]         destination:/dev/mqueue source:mqueue type:mqueue
2024/03/21 07:57:41 [adaptor/proxy]         destination:/sys source:sysfs type:sysfs
2024/03/21 07:57:41 [adaptor/proxy]         destination:/dev/shm source:/run/kata-containers/sandbox/shm type:bind
2024/03/21 07:57:41 [adaptor/proxy]         destination:/etc/resolv.conf source:/run/kata-containers/shared/containers/dc5d14f97adad43f08e9d30c8e93d400193a4b4904786f4776449e693a4dc847-317e09ac72c9a66f-resolv.conf type:bind
2024/03/21 07:57:41 [adaptor/proxy]     annotations:
2024/03/21 07:57:41 [adaptor/proxy]         io.kubernetes.cri.sandbox-id: dc5d14f97adad43f08e9d30c8e93d400193a4b4904786f4776449e693a4dc847
2024/03/21 07:57:41 [adaptor/proxy]         io.kubernetes.cri.sandbox-cpu-period: 100000
2024/03/21 07:57:41 [adaptor/proxy]         io.kubernetes.cri.sandbox-name: nginx
2024/03/21 07:57:41 [adaptor/proxy]         io.katacontainers.pkg.oci.container_type: pod_sandbox
2024/03/21 07:57:41 [adaptor/proxy]         io.kubernetes.cri.sandbox-uid: dd6cd64b-745a-48b1-a843-3a8dccfb7d34
2024/03/21 07:57:41 [adaptor/proxy]         nerdctl/network-namespace: /var/run/netns/cni-aca249c0-cf2a-69d2-ae44-be82d82f22ea
2024/03/21 07:57:41 [adaptor/proxy]         io.kubernetes.cri.sandbox-namespace: default
2024/03/21 07:57:41 [adaptor/proxy]         io.kubernetes.cri.sandbox-memory: 0
2024/03/21 07:57:41 [adaptor/proxy]         io.kubernetes.cri.container-type: sandbox
2024/03/21 07:57:41 [adaptor/proxy]         io.katacontainers.pkg.oci.bundle_path: /run/containerd/io.containerd.runtime.v2.task/k8s.io/dc5d14f97adad43f08e9d30c8e93d400193a4b4904786f4776449e693a4dc847
2024/03/21 07:57:41 [adaptor/proxy]         io.kubernetes.cri.sandbox-cpu-shares: 2
2024/03/21 07:57:41 [adaptor/proxy]         io.kubernetes.cri.sandbox-cpu-quota: 0
2024/03/21 07:57:41 [adaptor/proxy]         io.kubernetes.cri.sandbox-log-directory: /var/log/pods/default_nginx_dd6cd64b-745a-48b1-a843-3a8dccfb7d34
2024/03/21 07:57:41 [adaptor/proxy] getImageName: no pause image specified uses default pause image: registry.k8s.io/pause:3.7
2024/03/21 07:57:41 [adaptor/proxy] CreateContainer: calling PullImage for "registry.k8s.io/pause:3.7" before CreateContainer (cid: "dc5d14f97adad43f08e9d30c8e93d400193a4b4904786f4776449e693a4dc847")
2024/03/21 07:57:42 [adaptor/proxy] CreateContainer: successfully pulled image "registry.k8s.io/pause:3.7"
2024/03/21 07:57:42 [adaptor/proxy] StartContainer: containerID:dc5d14f97adad43f08e9d30c8e93d400193a4b4904786f4776449e693a4dc847
2024/03/21 07:57:53 [adaptor/proxy] CreateContainer: containerID:9da206395d8d3ed5153668d8d9489a17aed9a73d772a5406a3ffb9eb5250cdd3
2024/03/21 07:57:53 [adaptor/proxy]     mounts:
2024/03/21 07:57:53 [adaptor/proxy]         destination:/proc source:proc type:proc
2024/03/21 07:57:53 [adaptor/proxy]         destination:/dev source:tmpfs type:tmpfs
2024/03/21 07:57:53 [adaptor/proxy]         destination:/dev/pts source:devpts type:devpts
2024/03/21 07:57:53 [adaptor/proxy]         destination:/dev/mqueue source:mqueue type:mqueue
2024/03/21 07:57:53 [adaptor/proxy]         destination:/sys source:sysfs type:sysfs
2024/03/21 07:57:53 [adaptor/proxy]         destination:/sys/fs/cgroup source:cgroup type:cgroup
2024/03/21 07:57:53 [adaptor/proxy]         destination:/etc/config source:/run/kata-containers/shared/containers/9da206395d8d3ed5153668d8d9489a17aed9a73d772a5406a3ffb9eb5250cdd3-7d2a657d0b8035f6-config type:bind
2024/03/21 07:57:53 [adaptor/proxy]         destination:/sealed/etc/secret source:/run/kata-containers/shared/containers/9da206395d8d3ed5153668d8d9489a17aed9a73d772a5406a3ffb9eb5250cdd3-3ccecd05b5f6a69a-secret type:bind
2024/03/21 07:57:53 [adaptor/proxy]         destination:/etc/hosts source:/run/kata-containers/shared/containers/9da206395d8d3ed5153668d8d9489a17aed9a73d772a5406a3ffb9eb5250cdd3-ad6fa77b857f1074-hosts type:bind
2024/03/21 07:57:53 [adaptor/proxy]         destination:/dev/termination-log source:/run/kata-containers/shared/containers/9da206395d8d3ed5153668d8d9489a17aed9a73d772a5406a3ffb9eb5250cdd3-bbbc4927fd1a7dbe-termination-log type:bind
2024/03/21 07:57:53 [adaptor/proxy]         destination:/etc/hostname source:/run/kata-containers/shared/containers/9da206395d8d3ed5153668d8d9489a17aed9a73d772a5406a3ffb9eb5250cdd3-7da2f76697b03dc0-hostname type:bind
2024/03/21 07:57:53 [adaptor/proxy]         destination:/etc/resolv.conf source:/run/kata-containers/shared/containers/9da206395d8d3ed5153668d8d9489a17aed9a73d772a5406a3ffb9eb5250cdd3-eb04bd053187c277-resolv.conf type:bind
2024/03/21 07:57:53 [adaptor/proxy]         destination:/dev/shm source:/run/kata-containers/sandbox/shm type:bind
2024/03/21 07:57:53 [adaptor/proxy]         destination:/var/run/secrets/kubernetes.io/serviceaccount source:/run/kata-containers/shared/containers/9da206395d8d3ed5153668d8d9489a17aed9a73d772a5406a3ffb9eb5250cdd3-1bfe95e6daea0391-serviceaccount type:bind
2024/03/21 07:57:53 [adaptor/proxy]     annotations:
2024/03/21 07:57:53 [adaptor/proxy]         io.kubernetes.cri.image-name: docker.io/library/nginx:latest
2024/03/21 07:57:53 [adaptor/proxy]         io.kubernetes.cri.container-type: container
2024/03/21 07:57:53 [adaptor/proxy]         io.kubernetes.cri.sandbox-name: nginx
2024/03/21 07:57:53 [adaptor/proxy]         io.kubernetes.cri.sandbox-namespace: default
2024/03/21 07:57:53 [adaptor/proxy]         io.kubernetes.cri.container-name: nginx
2024/03/21 07:57:53 [adaptor/proxy]         io.kubernetes.cri.sandbox-uid: dd6cd64b-745a-48b1-a843-3a8dccfb7d34
2024/03/21 07:57:53 [adaptor/proxy]         io.katacontainers.pkg.oci.container_type: pod_container
2024/03/21 07:57:53 [adaptor/proxy]         io.kubernetes.cri.sandbox-id: dc5d14f97adad43f08e9d30c8e93d400193a4b4904786f4776449e693a4dc847
2024/03/21 07:57:53 [adaptor/proxy]         io.katacontainers.pkg.oci.bundle_path: /run/containerd/io.containerd.runtime.v2.task/k8s.io/9da206395d8d3ed5153668d8d9489a17aed9a73d772a5406a3ffb9eb5250cdd3
2024/03/21 07:57:53 [adaptor/proxy] getImageName: got image from annotations: docker.io/library/nginx:latest
2024/03/21 07:57:53 [adaptor/proxy] CreateContainer: calling PullImage for "docker.io/library/nginx:latest" before CreateContainer (cid: "9da206395d8d3ed5153668d8d9489a17aed9a73d772a5406a3ffb9eb5250cdd3")
2024/03/21 07:57:57 [adaptor/proxy] CreateContainer: successfully pulled image "docker.io/library/nginx:latest"
2024/03/21 07:57:58 [adaptor/proxy] StartContainer: containerID:9da206395d8d3ed5153668d8d9489a17aed9a73d772a5406a3ffb9eb5250cdd3
```

**Review by mkulke — APPROVED** [review]
I ran simple smoke tests using a CAA image built from this branch (launch + delete peerpods), did not observe any problems.

Reviewed the code as far as feasible: LGTM, great work!

**Comment by stevenhorsman** [comment]
Sorry if this comes across as bike shedding, but do you think some of the docs should remain in the root of the repo (under a new/old docs directory as they relate to the project as a whole and are more tricky to find after the move. I'm pariticularly thinking of the architecture and release process docs as they cover all the components, not just the CAA.

I think the golang-fedora image build is also not specific to the CAA? https://github.com/liudalibj/cloud-api-adaptor/blob/toplevel-cloud-module/src/cloud-api-adaptor/Dockerfile (it might even just be used in one of the controllers, so be a candidate to move into their directories, or be renamed for clarity I guess?

**Comment by liudalibj** [comment]
> Sorry if this comes across as bike shedding, but do you think some of the docs should remain in the root of the repo (under a new/old docs directory as they relate to the project as a whole and are more tricky to find after the move. I'm pariticularly thinking of the architecture and release process docs as they cover all the components, not just the CAA.
> 
> I think the golang-fedora image build is also not specific to the CAA? https://github.com/liudalibj/cloud-api-adaptor/blob/toplevel-cloud-module/src/cloud-api-adaptor/Dockerfile (it might even just be used in one of the controllers, so be a candidate to move into their directories, or be renamed for clarity I guess?

Good points, I updated the pr:
- keep the `architecture` and `release process` documents in the root docs hack folder
- keep golang-fedora docker file in the root hack folder

File: .github/workflows/azure-e2e-test.yml
Comment (stevenhorsman) [inline .github/workflows/azure-e2e-test.yml:1]: Nit:

Maybe updating the commit header to something like:
`workflows: Update workflows to reflect new directory structure`
might be better?

File: .github/workflows/caa_build_and_push.yaml [hunk 1]
```diff
@@ -4,7 +4,7 @@ on:
   workflow_call:
     inputs:
       registry:
-        default: 'quay.io/confidential-containers'
+        default: 'quay.io/liudalibj'
```
Comment (stevenhorsman) [inline .github/workflows/caa_build_and_push.yaml]: Should the registries in these workflows files be moved back to `quay.io/confidential-containers` from `quay.io/liudalibj`?

Comment (stevenhorsman) [reply .github/workflows/caa_build_and_push.yaml]: Ah, I've spotted these are reverted in a later commit. Maybe fixup 118f8f83decc494ed0276fb702a8addbae024ba5 into this one for better commit clarity?

**Review by stevenhorsman — COMMENTED** [review]
Just a couple of minor things I spotted scanning through the commits. Obviously there are too many changes to be confident for human review, so I'll try and run through a few builds as well.

File: .github/workflows/azure-e2e-test.yml
Comment (liudalibj) [reply .github/workflows/azure-e2e-test.yml:1]: updated.

File: .github/workflows/caa_build_and_push.yaml (hunk 1 above)
Comment (liudalibj) [reply .github/workflows/caa_build_and_push.yaml]: yeah fixuped the commit with others.

**Comment by stevenhorsman** [comment]
Hey @liudalibj - I was trying to test this out and hit a problem that I think is related to the directory shuffle - the undeploy of peerpod-ctrl seems to be failing in a way that I don't see in the previous code:
```
~/cloud-api-adaptor/src/cloud-api-adaptor# make delete
make -C ../peerpod-ctrl undeploy
make[1]: Entering directory '/root/cloud-api-adaptor/src/peerpod-ctrl'
/root/cloud-api-adaptor/src/peerpod-ctrl/bin/kustomize build config/default | kubectl delete --ignore-not-found=false -f -
bash: /root/cloud-api-adaptor/src/peerpod-ctrl/bin/kustomize: No such file or directory
No resources found
make[1]: *** [Makefile:182: undeploy] Error 127
make[1]: Leaving directory '/root/cloud-api-adaptor/src/peerpod-ctrl'
make: *** [Makefile:137: delete] Error 2
```
In my old code, after running the same thing, I can see that `./peerpod-ctrl/bin/kustomize` that been created.

**Comment by stevenhorsman** [comment]
> Hey @liudalibj - I was trying to test this out and hit a problem that I think is related to the directory shuffle - the undeploy of peerpod-ctrl seems to be failing in a way that I don't see in the previous code:
> 
> ```
> ~/cloud-api-adaptor/src/cloud-api-adaptor# make delete
> make -C ../peerpod-ctrl undeploy
> make[1]: Entering directory '/root/cloud-api-adaptor/src/peerpod-ctrl'
> /root/cloud-api-adaptor/src/peerpod-ctrl/bin/kustomize build config/default | kubectl delete --ignore-not-found=false -f -
> bash: /root/cloud-api-adaptor/src/peerpod-ctrl/bin/kustomize: No such file or directory
> No resources found
> make[1]: *** [Makefile:182: undeploy] Error 127
> make[1]: Leaving directory '/root/cloud-api-adaptor/src/peerpod-ctrl'
> make: *** [Makefile:137: delete] Error 2
> ```
> 
> In my old code, after running the same thing, I can see that `./peerpod-ctrl/bin/kustomize` that been created.

Oh - I think this is broken in `main` as well - I happened to be running it on Pradipta's docker provider fork where it worked.

**Review by huoqifeng — APPROVED** [review]
LGTM, fantastic! thank you @liudalibj

**Pull request merged**